src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java

```diff
@@ -620,6 +620,27 @@ public EnumActionResult onProcessRightClickBlock(PlayerInteractionManager intera
         return actionResult;
     }
 
+    @Redirect(method = "processTryUseItem", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/management/PlayerInteractionManager;processRightClick(Lnet/minecraft/entity/player/EntityPlayer;Lnet/minecraft/world/World;Lnet/minecraft/item/ItemStack;Lnet/minecraft/util/EnumHand;)Lnet/minecraft/util/EnumActionResult;"))
+    public EnumActionResult onProcessRightClick(PlayerInteractionManager interactionManager, EntityPlayer player, net.minecraft.world.World worldIn, @Nullable ItemStack stack, EnumHand hand) {
+        EnumActionResult actionResult = interactionManager.processRightClick(this.player, worldIn, stack, hand);
+        // If a plugin or mod has changed the item, avoid restoring
+        if (!SpongeCommonEventFactory.playerInteractItemChanged) {
```

Comment:
Is it possible to use some phase context flag that gets flipped here instead of checking a static variable? While I do want to standardize the interact events, I do want to see some of the legacy style static variable nonsense going the way of the dodo.

Reply:
Any chance this was done for performance reasons since this is a "hot" method ?

Reply:
It's blindingly fast to still have the check through the phase tracker, especially since it's still another boolean field, just a few field accesses. If it were say adding extra processing costs to mass convert and cast, then sure.

Reply:
Fair enough, I was playing devil's advocate